gnomad/utils/constraint.py

```diff
@@ -125,3 +127,274 @@ def collapse_strand(
+    :return: Coverage model and plateau models.
+    """
+    # Filter to sites with coverage above `cov_cutoff`
+    all_high_coverage_ht = coverage_ht.filter(coverage_ht.exome_coverage >= cov_cutoff)
+    agg_expr = {
+        "observed_variants": hl.agg.sum(all_high_coverage_ht.observed_variants),
+        "possible_variants": hl.agg.sum(all_high_coverage_ht.possible_variants),
+    }
+    for pop in pops:
+        agg_expr[f"observed_{pop}"] = hl.agg.array_sum(
+            all_high_coverage_ht[f"downsampling_counts_{pop}"]
+        )
+    high_coverage_ht = all_high_coverage_ht.group_by(*keys).aggregate(**agg_expr)
+    high_coverage_ht = annotate_mutation_type(high_coverage_ht)
+
+    # Build plateau models
+    plateau_models = build_plateau_models_total(
+        high_coverage_ht,
+        cpg="cpg",
+        mu_snp="mu_snp",
+        observed_variants="observed_variants",
+        possible_variants="possible_variants",
+        weighted=weighted,
+    )
+    # Build plateau models for populations if `pops` is given
+    if pops:
+        pop_lengths = get_all_pop_lengths(high_coverage_ht, pops)
+        for length, pop in pop_lengths:
+            pop_plateau_model = build_plateau_models_pop(
+                high_coverage_ht,
+                cpg="cpg",
+                mu_snp="mu_snp",
+                observed_variants=f"observed_{pop}",
+                possible_variants="possible_variants",
+                pop=pop,
+                length=length,
+                weighted=weighted,
+            )
+            plateau_models = plateau_models.annotate(**pop_plateau_model[pop])
+
+    # Build coverage model
+    all_low_coverage_ht = coverage_ht.filter(
+        (coverage_ht.exome_coverage < cov_cutoff) & (coverage_ht.exome_coverage > 0)
+    )
+    # a metric that represents the relative mutability of the exome for high coverage sites as a global scaling factor
+    high_coverage_scale_factor = all_high_coverage_ht.aggregate(
+        hl.agg.sum(all_high_coverage_ht.observed_variants)
+        / hl.agg.sum(
+            all_high_coverage_ht.possible_variants * all_high_coverage_ht.mu_snp
+        )
+    )
+    low_coverage_ht = all_low_coverage_ht.group_by(
+        log_coverage=hl.log10(all_low_coverage_ht.exome_coverage)
+    ).aggregate(
+        low_coverage_obs_exp=hl.agg.sum(all_low_coverage_ht.observed_variants)
+        / (
+            high_coverage_scale_factor
+            * hl.agg.sum(
+                all_low_coverage_ht.possible_variants * all_low_coverage_ht.mu_snp
+            )
+        )
+    )
+    coverage_model = build_coverage_model(
+        low_coverage_ht,
+        low_coverage_obs_exp="low_coverage_obs_exp",
+        log_coverage="log_coverage",
+    )
+    # TODO: consider weighting here as well
```

Comment:
```suggestion
    # TODO: consider weighting here as well
    coverage_model = build_coverage_model(
        low_coverage_ht,
        low_coverage_obs_exp="low_coverage_obs_exp",
        log_coverage="log_coverage",
    )
```
moving the comment to make it clearer